### PR TITLE
:tada: [feat]: Tooltip to display absolute time when using relative date and time format (#13764)

### LINKED ISSUE #13764: Always display the absolute time in a tooltip when using the relative date and time format
### Problem to solve

We've been using **relative** and **absolute** date and time formats across the product. For example, see [projects](https://gitpod.io/t/gitpod/projects), [members](https://gitpod.io/t/gitpod/members), and [usage](https://gitpod.io/t/gitpod/usage) for the Gitpod team in Gitpod.io.

Generally, both absolute and relative formats can be useful depending on each use case. 💡 

:a: Relative date and time can be useful when the content is updated frequently and surfacing an absolute date is not critical, etc.
:b: Absolute date and time makes sense when users would need to convert the date in order to complete a task, etc.

### Proposal

In any case, I think we can safely use the following guideline across the product:

> When using the relative format, always display the absolute time in a tooltip.

This issue acts as a placeholder for adding a tooltip on all relative date and time formats used across the product. 🛹 

### Next steps

Future next steps include:
1. Documenting this as a guideline.
2. Introducing more sophisticated guidelines on when to use which.
3. Explain in detail date and time format options and how to use them.

### PR DESCRIPTION
**Pull request by tanishq-singh-2407**
## Description
- Display the `absolute time` in a tooltip when using the `relative date and time` format.
- Proxied `/public-api` to `DEV_HOST`.

|_**Before**_|_**After**_|
|-------------|------------|
|Screenshot (388)|Screenshot (389)|

## Related Issue(s)

Fixes #13764 

## How to test

- Hover mouse over any `relative date and time` format.

## Release Notes

```release-note
Display a tolltip with absolute date when using relative date format
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [x] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/dashboard/src/projects/Project.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Project.tsx:375]: **issue**: This is causing some duplicate elements to show up on the project **branches** and **prebuilds** pages.


| Branches| Prebuilds |
|-|-|
| Screenshot 2023-01-24 at 12 08 00 AM (2) | Screenshot 2023-01-24 at 12 07 56 AM (2) |

File: components/dashboard/src/teams/Members.tsx
Comment (gtsiolis) [inline components/dashboard/src/teams/Members.tsx:242]: issue: Did we change any hover style on the members page?

| Members (Default) | Members (Hover) |
|-|-|
| Screenshot 2023-01-24 at 12 09 30 AM (2) | Screenshot 2023-01-24 at 12 09 32 AM (2) |

File: components/dashboard/src/workspaces/WorkspaceEntry.tsx
Comment (gtsiolis) [inline components/dashboard/src/workspaces/WorkspaceEntry.tsx:209]: **question**: Is this working on the workspaces list? Seems like the tooltip duplicates what's on the date below.

| Workspaces |
|-|
| Screenshot 2023-01-24 at 12 11 39 AM (2) |

File: components/dashboard/craco.config.js
Comment (gtsiolis) [inline components/dashboard/craco.config.js:34]: **question**: Is this needed here? Not sure but I'll let someone from the `@gitpod-io/engineering-webapp`  take a look.

**Review by gtsiolis — CHANGES_REQUESTED** [review]
Thanks for opening and your patience, @tanishq-singh-2301!

Left some UX comments below that we should address before merging this. 🏁 

It seems that some pages in the admin dashboard still lack the tooltip with the absolute timestamp. Could you add these, too? See screenshot below. If it's too much to ask let's open a follow up issue or PR to keep track of this. 🏓 

| Admin / Users | Admin / Workspaces | Admin / Projects |
|-|-|-|
| Screenshot 2023-01-24 at 12 13 38 AM (2) | Screenshot 2023-01-24 at 12 13 50 AM (2) | Screenshot 2023-01-24 at 12 13 54 AM (2) |

💡 **tip**: You could re-open the previous PR by pushing more commits but opening a new PR is also fine.

Re-added also the **~cla: signed** label from https://github.com/gitpod-io/gitpod/pull/15812 for visibility for the rest of the reviewers.

File: components/dashboard/craco.config.js
Comment (tanishq-singh-2301) [reply components/dashboard/craco.config.js:34]: Sure!!

**Comment by gtsiolis** [comment]
@tanishq-singh-2301 Could you resolve the conflicts here?

/werft run

File: components/dashboard/src/projects/Project.tsx
Comment (gtsiolis) [reply components/dashboard/src/projects/Project.tsx:375]: Looks good—resolving.

File: components/dashboard/src/teams/Members.tsx
Comment (gtsiolis) [reply components/dashboard/src/teams/Members.tsx]: Looks good—resolving.

File: components/dashboard/src/workspaces/WorkspaceEntry.tsx
Comment (gtsiolis) [reply components/dashboard/src/workspaces/WorkspaceEntry.tsx]: Looks good—resolving.

File: components/dashboard/src/admin/ProjectsSearch.tsx
Comment (gtsiolis) [inline components/dashboard/src/admin/ProjectsSearch.tsx:177]: **suggestion**: I skipped mentioning in https://github.com/gitpod-io/gitpod/pull/15994#pullrequestreview-1266443010 that this is also missing from the teams search in the admin dashboard. Sorry for not clarifying—if that's trivial for you to add could you please add it there, too? Otherwise, we can merge this UX as is an open a follow-up PR.

**Review by gtsiolis — APPROVED** [review]
Thanks again, @tanishq-singh-2301!

UX LGTM! 🏁 

@geropl Could you take a look or pass this to someone on the team for taking a look at this question (https://github.com/gitpod-io/gitpod/pull/15994#pullrequestreview-1266443010) regarding a code changes and whether it's needed?

/hold

File: components/dashboard/src/admin/ProjectsSearch.tsx
Comment (tanishq-singh-2301) [reply components/dashboard/src/admin/ProjectsSearch.tsx:177]: Actually I can't access `admin` route, I can add tooltip but I'll not sure if it's gonna fit or break something else.

**Comment by gtsiolis** [comment]
/werft run with-clean-slate-deployment=true with-preview=true recreate-vm=true

File: components/dashboard/src/admin/ProjectsSearch.tsx
Comment (gtsiolis) [reply components/dashboard/src/admin/ProjectsSearch.tsx:177]: @tanishq-singh-2301 It's the same area where you've made the same changes for [projects](https://github.com/gitpod-io/gitpod/pull/15994/files#diff-6fa0733647f6ea4dc958623adcead0f2cec8c9b64a26406b7a77b1d20bddd624), [users](https://github.com/gitpod-io/gitpod/pull/15994/files#diff-289e945ddd7cbc2f8d552b92792839d7329dcd6f7001c6600136aee9bb2f32c9), and [workspaces](https://github.com/gitpod-io/gitpod/pull/15994/files#diff-caceff449f575088aad98c86d4e9d600f66db721517693f332beabaf3bd9bec5) in the admin dashboard. However, this should be good to merge as is. Let's wait for someone from `@gitpod-io/engineering-webapp` to chime in for the final approval.

**Comment by tanishq-singh-2301** [comment]
any update?

**Comment by gtsiolis** [comment]
@tanishq-singh-2301 Could you resolve the conflicts once more here? By the way, this is one good example why working in smaller ([MVC](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#0c2e0d6c8a99404b97ccdc206b573d72)) PRs is better as you run into conflicting files much less frequently. It's ok in this case—it just happened to write on top of frequently changed files. 🏓 

Let me ping the @gitpod-io/engineering-webapp for visibility and spin up the preview environment once more.

/werft run with-clean-slate-deployment=true with-preview=true recreate-vm=true

**Comment by gtsiolis** [comment]
/werft run with-clean-slate-deployment=true with-preview=true recreate-vm=true

**Comment by tanishq-singh-2301** [comment]
I tried doing that but I'm not able to squash all the commit, as there are many commits from other developers (in between my commits) that I merged from main to keep the branch up to date.

I saw some yt videos and read some articles on how to rebase commits and found this: `git rebase -i HEAD~9`, but In my case there are like 416 commits from other dev in between mine 9 commits.

What to do?

**Comment by gtsiolis** [comment]
@tanishq-singh-2301 Sometimes rebase can be challenging when multiple commits have changed the same lines. The easiest route could be to create a new branch with the same name and apply all changes one by one of the latest changes from `main`. 💡

**Pull request closed without merging**
